diff -rup Sys-Virt-TCK-0.1.0.orig/conf/default.cfg Sys-Virt-TCK-0.1.0.new/conf/default.cfg
--- Sys-Virt-TCK-0.1.0.orig/conf/default.cfg	2009-07-22 18:33:55.000000000 +0100
+++ Sys-Virt-TCK-0.1.0.new/conf/default.cfg	2009-08-26 16:17:10.000000000 +0100
@@ -54,8 +54,8 @@ kernels = (
        hvm
        xen
      )
-     kernel = http://download.fedora.redhat.com/pub/fedora/linux/releases/11/Fedora/i386/os/images/pxeboot/vmlinuz
-     initrd = http://download.fedora.redhat.com/pub/fedora/linux/releases/11/Fedora/i386/os/images/pxeboot/initrd.img
+     kernel = http://download.fedora.redhat.com/pub/fedora/linux/releases/11/Fedora/i386/os/images/pxeboot/vmlinuz-PAE
+     initrd = http://download.fedora.redhat.com/pub/fedora/linux/releases/11/Fedora/i386/os/images/pxeboot/initrd-PAE.img
    }
    # Fedora 11 x86_64 has pv_ops, so one kernel can do both Xen and KVM guests here
    {
