diff -rup Sys-Virt-TCK-0.1.0.orig/lib/Sys/Virt/TCK.pm Sys-Virt-TCK-0.1.0.new/lib/Sys/Virt/TCK.pm
--- Sys-Virt-TCK-0.1.0.orig/lib/Sys/Virt/TCK.pm	2009-08-26 16:13:28.000000000 +0100
+++ Sys-Virt-TCK-0.1.0.new/lib/Sys/Virt/TCK.pm	2009-08-26 16:16:19.000000000 +0100
@@ -72,7 +72,7 @@ sub setup {
 sub sanity_check {
     my $self = shift;
 
-    my @doms = $self->{conn}->list_domains;
+    my @doms = grep { $_->get_id > 0 } $self->{conn}->list_domains;
     if (@doms) {
 	die "there is/are " . int(@doms) . " pre-existing active domain(s) in this driver";
     }
@@ -96,11 +96,9 @@ sub sanity_check {
 sub reset {
     my $self = shift;
 
-    my @doms = $self->{conn}->list_domains;
+    my @doms = grep { $_->get_id > 0 } $self->{conn}->list_domains;
     foreach my $dom (@doms) {
-	if ($dom->get_id != 0) {
-	    $dom->destroy;
-	}
+	$dom->destroy;
     }
 
     @doms = $self->{conn}->list_defined_domains();
